playwright/_impl/_network.py

```diff
@@ -262,6 +266,8 @@ def _target_closed_future(self) -> asyncio.Future:
         return page._closed_or_crashed_future
 
     def _safe_page(self) -> "Optional[Page]":
+        if self.from_service_worker:
+            return None
         return cast("Frame", from_channel(self._initializer["frame"]))._page
```

Comment:
```suggestion
        frame = from_nullable_channel(self._initializer["frame"])
        if not frame:
            return None
        return cast("Frame", frame)._page
```

maybe something like this instead? This would align it with the upstream port: 

https://github.com/microsoft/playwright/blob/9ca895dea087a36219202624a8550921bb924bf0/packages/playwright-core/src/client/network.ts#L219

Reply:
`self._initializer["frame"]` this will throw `"KeyError" `
I can do something like this:
```
try:
         frame = from_nullable_channel(self._initializer["frame"])
except KeyError:
         return None
return cast("Frame", frame)._page
```

Reply:
Let's do:

```python
   frame = from_nullable_channel(self._initializer.get("frame"))
   if not frame:
       return None
   return cast("Frame", frame)._page
```